### ISSUE TITLE
env-vars overrides applied to FunctionName, not resource IDs

### ISSUE THREAD

**Issue by mbradburn** [awslabs/aws-sam-cli#1888]
### Description

With sam local invoke, it used to be that the environment variable overrides in the --env-vars file were keyed by the function's resource ID.. the env-vars file contained a mapping from a function Resource ID to the environment variable overrides for that function.  They're apparently now a mapping from the function's FunctionName property to its environment variables. 

### Steps to reproduce

Provide steps to replicate.

### Observed result

When my env-vars file is keyed by function resource ID, the environment variables are not present in the environment of the locally running function.

### Expected result

Expected the environment variables to be overridden.

1. OS:
Darwin mjb-2.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64

2. `sam --version`:

SAM CLI, version 0.46.1

`Add --debug flag to command you are running`

... let me know if you need more info.

**Comment by jfuss** [comment]
@mbarton Thanks for the report. For now you will need to pin to 0.45.0. In 0.46.0, we added the ability to invoke with a function name instead of the LogicalId. There were some changes made in that PR that caused this issue. I have a working patch locally right now. I need to go back through the original PR (tomorrow) and make sure we fix everything (we patched a different issue related to the same thing earlier today). Sorry for the troubles and regression!
